sigstore/_verify/policy.py

```diff
@@ -0,0 +1,233 @@
+# Copyright 2022 The Sigstore Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+APIs for describing identity verification "policies", which describe how the identities
```

Comment:
I think we need to expand on this a bit. I understand what the API is getting at since I'm reviewing the code. But if I was just looking at docs and saw a function that takes a "verification policy", I'd be pretty confused.

Reply:
100% agree. I've been thinking more generally about a mental model for Sigstore verifications, with three moving parts:

* "Verification context": the immutable and invariant things needed to *perform* a verification (the root cert(s), cert chain(s), various instance URLs, etc.). These are composed in the `Verifier` instance itself.
* "Verification materials": the principals being verified: input, signature, cert, hash, and (maybe) an offline Rekor entry. These are composed in the `VerificationMaterials` model.
* "Verification policy": *how* the verification materials are verified, beyond their basic cryptographic authenticity/integrity: the identity/context they relate to, etc. These are composed via the "policies" API, and can be mixed together via `AnyOf` and `AllOf`.

Does that general categorization make sense to you? If so I can put that into the source, in each of the relevant modules 🙂

Reply:
Yep, that makes a lot of sense to me.